### ISSUE TITLE
Package validation fail in .NET 8 with Unnecessary suppressions found

### ISSUE THREAD

**Issue by McMlok** [dotnet/sdk#34936]
### Describe the bug
When I switch SDK to 8.0.100-preview.7.23376.3 I'm not able to pack my packages with error 

"Unnecessary suppressions found. The APICompat suppression file can be updated by
 rebuilding with '/p:ApiCompatGenerateSuppressionFile=true'"

I think an issue is in different check in CP0007 and CP0002 rules between SDK 7.0.400 and 8.0.100-preview.7.23376.3. With .NET 7 SDK it does not recognise that base class is located in referenced nuget package which is not included in actual solution and mark that property is missing in new version. But in .NET 8 this works and in suppression file are unnecessary records. This behavior breaks our work when some developers in a team already play with .NET 8 and others and also build servers are still using .NET 7 SDK and we are not able to have same content in suppression file for all. 

I know workaround is to explicitly specify SDK version in global.json but change behavior of base line validator to not throw error when there are unnecessary suppressions and instead report them as warning will help.

### To Reproduce
- download [repro](https://github.com/McMlok/PackValidationTest/tree/master)
- set SOME_LOCAL_FOLDER to your nuget sources
- create nuget package to your local nuget feed with command _dotnet pack . --output SOME_LOCAL_FOLDER_ and  SDK .NET 7.0.400
- switch to branch [newVersion ](https://github.com/McMlok/PackValidationTest/tree/newVersion)
- change used SDK in global.json to 8.0.100-preview.7.23376.3
- create new version of the nuget package with enable baseline validation with command _dotnet pack . --output SOME_LOCAL_FOLDER_

Thank you

**Comment by ViktorHofer** [comment]
Hello @McMlok, thanks for reaching out.

This feature was added with https://github.com/dotnet/sdk/commit/a3303b614e235dea5d90283bb8fcc09d5bfcb86f. You can disable the behavior by setting the following two properties to true: https://github.com/dotnet/sdk/blob/a73ef0378f1eef9c4abf00d701ddc5acdac757cf/src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ApiCompat.ValidatePackage.targets#L52-L53

```
<PropertyGroup>
  <ApiCompatPreserveUnnecessarySuppressions>true</ApiCompatPreserveUnnecessarySuppressions>
  <ApiCompatPermitUnnecessarySuppressions>true</ApiCompatPermitUnnecessarySuppressions>
</PropertyGroup>
```

We need to document this feature as part of the .NET 8 release. Please let me know if that works for you.

**Comment by McMlok** [comment]
Yes, your solution works, thank you. Can I close the issue or you want as remainder to enhance the documentation? From my side I have what I wanted :-).
